components/audit_table/user_row/index.ts

```diff
@@ -0,0 +1,140 @@
+        desc = intl.formatMessage(holders.updateGeneral);
+        break;
+    case '/users/newpassword':
+        if (userInfo[0] === 'attempted') {
+            desc = intl.formatMessage(holders.attemptedPassword);
+        } else if (userInfo[0] === 'completed') {
+            desc = intl.formatMessage(holders.successfullPassword);
+        } else if (
+            userInfo[0] ===
+                'failed - tried to update user password who was logged in through oauth'
+        ) {
+            desc = intl.formatMessage(holders.failedPassword);
+        }
+
+        break;
+    case '/users/update_roles': {
+        const userRoles = userInfo[0].split('=')[1];
+
+        desc = intl.formatMessage(holders.updatedRol);
+        if (userRoles.trim()) {
+            desc += userRoles;
+        } else {
+            desc += intl.formatMessage(holders.member);
+        }
+
+        break;
+    }
+    case '/users/update_active': {
+        const updateType = userInfo[0].split('=')[0];
+        const updateField = userInfo[0].split('=')[1];
+
+        /* Either describes account activation/deactivation or a revoked session as part of an account deactivation */
+        if (updateType === 'active') {
+            if (updateField === 'true') {
+                desc = intl.formatMessage(holders.accountActive);
+            } else if (updateField === 'false') {
+                desc = intl.formatMessage(holders.accountInactive);
+            }
+
+            const actingUserInfo = userInfo[1].split('=');
+            if (actingUserInfo[0] === 'session_user') {
+                const actingUser = getUser(state, actingUserInfo[1]);
+                const user = getCurrentUser(state);
+                if (user && actingUser && isSystemAdmin(user.roles)) {
+                    desc += intl.formatMessage(holders.by, {
+                        username: actingUser.username,
+                    });
+                } else if (user && actingUser) {
+                    desc += intl.formatMessage(holders.byAdmin);
+                }
+            }
+        } else if (updateType === 'session_id') {
+            desc = intl.formatMessage(holders.sessionRevoked, {
+                sessionId: updateField,
+            });
+        }
+
+        break;
+    }
+    case '/users/send_password_reset':
+        desc = intl.formatMessage(holders.sentEmail, {
+            email: userInfo[0].split('=')[1],
+        });
+        break;
+    case '/users/reset_password':
+        if (userInfo[0] === 'attempt') {
+            desc = intl.formatMessage(holders.attemptedReset);
+        } else if (userInfo[0] === 'success') {
+            desc = intl.formatMessage(holders.successfullReset);
+        }
+
+        break;
+    case '/users/update_notify':
+        desc = intl.formatMessage(holders.updateGlobalNotifications);
+        break;
+    default:
+        break;
+    }
+
+    return {
+        desc,
+    };
+}
+
+export default connect(mapStateToProps)(AuditRow);
```

Comment:
I understand how it works, but looks a bit weird to me. I would expect to find here the `UserRow` component (same for the `ChannelRow`).

Also another tiny optimization here, is the fact that you are generating the description every single `mapStateToProps`, that gets executed every store update (nothing super critical, but can be improved just passing the values to the component and and letting the component to generate the final description on render.

Reply:
I fixed to generate the final description on render.
I think the better way to implement this is using hooks so I used hooks instead of connect.